### ISSUE TITLE
Bibliography repeated twice when using citeproc explicitely

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#5585]
This is a regression in 1.3 as it was fixed in 1.2 by @dragonstyle already https://github.com/quarto-dev/quarto-cli/issues/2393

* References.bib
```
@article{Cronbach_1951,
	doi = {10.1007/bf02310555},
	year = 1951,
	month = {sep},
	publisher = {Springer Science and Business Media {LLC}},
	vol = {16},
	number = {3},
	pages = {297--334},
	author = {Lee J. Cronbach},
	title = {Coefficient alpha and the internal structure of tests},
	journal = {Psychometrika}
}
```

```markdown
---
title: "Citeproc"
format: html
bibliography: References.bib
citeproc: true
---

Hello [@Cronbach_1951]
```
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/c0e737af-9416-43e6-8207-36407b5d0c2c)

We have a code path where `citeproc: true` should be removed : 
https://github.com/quarto-dev/quarto-cli/blob/e136ef277275a9ad9d4ff4599b04595c34c4f4ef/src/command/render/filters.ts#L619-L624

but it seems to still be present in the default file we generate.

This is happening with current release 1.3 and pre release 1.4. 

So This is a regression we should probably patch for 1.3 and fix for 1.4 - this is needed to work for #5575 as we set `citeproc: false` by default, and we need user to set `citeproc: true` to change the engine.

**Comment by cderv** [comment]
@cscheid probably a side effect of this change I made as `resolvedFilters()` happens now on a cloned version of options 
https://github.com/quarto-dev/quarto-cli/blob/8781ac3187fe937e9e42916d1e99fd73a6c7c6d9/src/command/render/defaults.ts#L43-L53

This change was done in https://github.com/quarto-dev/quarto-cli/pull/4348 to fix https://github.com/quarto-dev/quarto-cli/issues/4260 after a discussion with @dragonstyle

**Comment by cscheid** [comment]
If the breakage happened from 1.2 to 1.3, I think the issue is the difference between

https://github.com/quarto-dev/quarto-cli/blob/v1.2.475/src/resources/filters/quarto-pre/quarto-pre.lua

and

https://github.com/quarto-dev/quarto-cli/blob/v1.3.353/src/resources/filters/main.lua#L329-L348 ...

**Comment by cderv** [comment]
@cscheid I confirm it was what I mentioned. I have a PR with a fix in #5588 - no need to look at the Lua chain
